### PR TITLE
tikv-configuration-file: add config for compaction guard

### PR DESCRIPTION
**Pull request by CharLotteiu**
### What is changed, added or deleted? (Required)

- Add config for compaction guard.
- Fix a file format issue.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/4420
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tikv-configuration-file.md [hunk 1]
```diff
@@ -924,7 +924,7 @@ bloom filter 为每个 key 预留的长度。
 
 ### `target-file-size-base`
 
-+ base level 的目标文件大小。
++ base level 的目标文件大小。当 `enable-compaction-guard` 设为 `true` 时，`compaction-guard-max-output-file-size` 的值会覆盖此配置项.
```
Comment (TomShawn) [inline tikv-configuration-file.md:927]:
```suggestion
+ base level 的目标文件大小。当 `enable-compaction-guard` 的值为 `true` 时，`compaction-guard-max-output-file-size` 会覆盖此配置。
```

**Review by TomShawn — COMMENTED** [review]
@yiwu-arbug Please help take a look, thanks!

File: tikv-configuration-file.md [hunk 2]
```diff
@@ -998,6 +998,23 @@ Compaction 优先类型，默认：3（MinOverlappingRatio），0（ByCompensate
 + 默认值：256GB
 + 单位：KB|MB|GB
 
+### `enable-compaction-guard`
+
++ 设置 compaction guard 的启用状态。compaction guard 可以优化 TiKV Region 边界的 SST 文件分割，帮助降低 compaction I/O，让 TiKV 能够输出较大的 SST 文件，并且在迁移 Region 时及时清理过期数据。
```
Comment (yiwu-arbug) [inline tikv-configuration-file.md:1003]:
```suggestion
+ 设置 compaction guard 的启用状态。compaction guard 优化通过使用 TiKV Region 边界分割 SST 文件，帮助降低 compaction I/O，让 TiKV 能够输出较大的 SST 文件，并且在迁移 Region 时及时清理过期数据。
```

**Review by yiwu-arbug — COMMENTED** [review]
rest LGTM

File: tikv-configuration-file.md [hunk 3]
```diff
@@ -1094,6 +1111,23 @@ rocksdb writecf 相关的配置项。
 + 开启将整个 key 放到 bloom filter 中的开关。
 + 默认值：false
 
+### `enable-compaction-guard`
+
++ 设置 compaction guard 的启用状态。compaction guard 可以优化 TiKV Region 边界的 SST 文件分割，帮助降低 compaction I/O，让 TiKV 能够输出较大的 SST 文件，并且在迁移 Region 时及时清理过期数据。
```
Comment (TomShawn) [inline tikv-configuration-file.md:1116]:
```suggestion
+ 设置 compaction guard 的启用状态。compaction guard 优化通过使用 TiKV Region 边界分割 SST 文件，帮助降低 compaction I/O，让 TiKV 能够输出较大的 SST 文件，并且在迁移 Region 时及时清理过期数据。
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**